lib/elixir/lib/version.ex

```diff
@@ -402,9 +402,10 @@ defmodule Version do
     def parse_version(string, approximate? \\ false) when is_binary(string) do
       destructure [version_with_pre, build], String.split(string, "+", parts: 2)
       destructure [version, pre], String.split(version_with_pre, "-", parts: 2)
-      destructure [major, minor, patch], String.split(version, ".")
+      destructure [major, minor, patch, trailing], String.split(version, ".")
 
-      with {:ok, major} <- require_digits(major),
+      with nil <- trailing,
```

Comment:
Feedback on how do this better is appreciated. Would be nice if `destructure` somehow worked inside `with`.